Add a warning about upgrading tiflash using tiup (#8191)
pingcap/docs: upgrade-tidb-using-tiup.md

- Upgrade from TiDB 5.1 versions to TiDB 5.3 versions.
- Upgrade from TiDB 5.2 versions to TiDB 5.3 versions.

> **Warning:**
>

> > - You cannot upgrade TiFlash from versions earlier than v5.3 to v5.3 or later online. To achieve such a cross-version upgrade, you need to stop TiFlash instances of the early version first, and then upgrade the TiFlash instances. For details, see warnings in [Online upgrade](#online-upgrade).

Author: ```suggestion
> - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. To achieve such a cross-version upgrade, you need to stop TiFlash instances of the early version first, and then upgrade the TiFlash instances. For details, see warnings in [Online upgrade](#online-upgrade).
```

> > - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. Instead, you must first stop all the TiFlash instances of the early version, and then upgrade the cluster by following the upgrade process of the target version. For details, see warnings in [Online upgrade](#online-upgrade).

Author: ```suggestion
> - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. Instead, you must first stop all the TiFlash instances of the early version, and then upgrade the cluster offline. If other components (such as TiDB and  TiKV) do not support an online upgrade, follow the instructions in warnings in [Online upgrade](#online-upgrade).
```

> > - During an upgrade of TiFlash from earlier versions to v5.3, query errors may occur. To avoid this issue, run `set global tidb_allow_mpp = 0` and start the upgrade after the completion of all queries in the MPP mode. After TiFlash is upgraded to v5.3, run `set global tidb_allow_mpp = 1` to enable the MPP mode.

Reviewer: Generally, we'd suggest using "might" instead of "may", because "may" has multiple meanings.
```suggestion
> - During an upgrade of TiFlash from earlier versions to v5.3, query errors might occur. To avoid this issue, run the `set global tidb_allow_mpp = 0` command and start the upgrade after all queries in the MPP mode are completed. After TiFlash is upgraded to v5.3, run the `set global tidb_allow_mpp = 1` command to enable the MPP mode.
```

> > - During an upgrade from earlier versions to v5.3, query errors may occur. To avoid this issue, you can set `set global tidb_allow_mpp = 0` and start the upgrade after completion of all queries in the MPP mode. After all components are upgraded to v5.3, you can set `set global tidb_allow_mpp = 1` to enable the MPP mode.

Author: ```suggestion
> - During an upgrade of TiFlash from earlier versions to v5.3, query errors may occur. To avoid this issue, run `set global tidb_allow_mpp = 0` and start the upgrade after the completion of all queries in the MPP mode. After TiFlash is upgraded to v5.3, run `set global tidb_allow_mpp = 1` to enable the MPP mode.
```

> > - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. Instead, you must first stop all the TiFlash instances of the early version, and then upgrade the cluster offline. If other components (such as TiDB and  TiKV) do not support an online upgrade, follow the instructions in warnings in [Online upgrade](#online-upgrade).

Author: ```suggestion
> - You cannot upgrade TiFlash online from versions earlier than 5.3 to 5.3 or later. Instead, you must first stop all the TiFlash instances of the early version, and then upgrade the cluster offline. If other components (such as TiDB and  TiKV) do not support an online upgrade, follow the instructions in warnings in [Online upgrade](#online-upgrade).
```

> **Note:**
>
> + An online upgrade upgrades all components one by one. During the upgrade of TiKV, all leaders in a TiKV instance are evicted before stopping the instance. The default timeout time is 5 minutes (300 seconds). The instance is directly stopped after this timeout time.
>
> + To perform the upgrade immediately without evicting the leader, specify `--force` in the command above. This method causes performance jitter but not data loss.
>
> + To keep a stable performance, make sure that all leaders in a TiKV instance are evicted before stopping the instance. You can set `--transfer-timeout` to a larger value, for example, `--transfer-timeout 3600` (unit: second).
>

> > - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. Instead, you must first stop all the TiFlash instances and then upgrade the cluster offline. Then, reload the cluster so that other components are upgraded online without interruption.

Author: ```suggestion
> - When upgrading a TiDB cluster from versions earlier than 5.3 to 5.3 or later, you cannot upgrade TiFlash online. Instead, you must first stop all the TiFlash instances and then upgrade the cluster offline. Then, reload the cluster so that other components are upgraded online without interruption.
```

> > - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. To achieve such a cross-version upgrade, you need to stop the TiFlash instances first and upgrade the cluster offline. Then, reload the cluster so that other components are upgraded online without interruption.

Author: ```suggestion
> - You cannot upgrade TiFlash online from versions earlier than v5.3 to v5.3 or later. Instead, you can stop the TiFlash instances first and upgrade the cluster offline. Then, reload the cluster so that other components are upgraded online without interruption.
```
